src/System/Random.hs

```diff
@@ -197,9 +200,15 @@ class Random a where
   -- | Takes a range /(lo,hi)/ and a pseudo-random number generator
   -- /g/, and returns a pseudo-random value uniformly distributed over the
   -- closed interval /[lo,hi]/, together with a new generator. It is unspecified
-  -- what happens if /lo>hi/. For continuous types there is no requirement
-  -- that the values /lo/ and /hi/ are ever produced, but they may be,
-  -- depending on the implementation and the interval.
+  -- what happens if /lo>hi/, but usually the values will simply get swapped.
+  --
+  -- For continuous types there is no requirement that the values /lo/ and /hi/ are ever
+  -- produced, but they may be, depending on the implementation and the interval.
+  --
+  -- There is no requirement to follow the @Ord@ instance and the concept of range can be
+  -- defined on per type basis. For example product types will treat their values
+  -- independently. In case when a lawful range is desired `uniformR` should be used
```

Comment:
It is a good place to elaborate what we mean by "independently" here. I think it's worth to give an example: "randomR (a,b) (c,d) generates tuples with the first component between a and c and the second component between b and d" or something along these lines.

Reply:
I've added a couple of examples. I am not sure I have enough brainpower to elaborate what "independently" means. If anyone feels like improving the haddock we can always accept a PR later.